### DISCUSSION TITLE
hi, does itwin.js currently support collision detection?

### DISCUSSION THREAD

**Discussion by shenshengyi** [iTwin/itwinjs-core#371] [Q&A] [answered]
hi. I would like to ask, does itwin.js currently support common collision detection in 3D?

**Comment by pmconne** [comment, marked as answer]
The [look-and-move](https://github.com/imodeljs/imodeljs/blob/c033f09315c7e1ce61ba48fe16c4f33eb7c28330/core/frontend/src/tools/ViewTool.ts#L1888) tool can be [configured](https://github.com/imodeljs/imodeljs/blob/c033f09315c7e1ce61ba48fe16c4f33eb7c28330/core/frontend/src/tools/ToolSettings.ts#L46) to detect collisions with walls and floor.
The collision detection is currently based on the direction in which you are looking - so if you're walking down a flight of stairs while looking up at the ceiling, it will not detect the floor (you won't move downward).

**Reply by shenshengyi** [reply]
OK, thank you very much for your answer. I'm going to use and learn the look and move tool.
